src/store/login/index.js

```diff
@@ -0,0 +1,178 @@
+import { createAction, handleActions } from 'redux-actions';
+import {
+  put, takeLatest, call, select
+} from 'redux-saga/effects';
+import { createSelector } from 'reselect';
+import history from '../../side-effects/history';
+import { setAppWaiting } from '../app';
+import authenticate from '../../api/auth';
+import getUserData from '../../api/users';
+
+/**
+ * INITIAL STATE
+ */
+
+export const initialState = {
+  jwtToken: localStorage.getItem('jwtToken') || '',
```

Comment:
This is not valid. The initial state should be ALWAYS the same and pure. In this case this is not reliable or pure.
As far as I can see you would like to get the jwtToken to see that is the user logged in or not. To achieve this, you should create a standalone saga that will check that for you and will call the proper actions which will set the state for you. That checker saga shoudl be called as soon as possible in the app flow.

Reply:
I see. Changed it as we discussed it